Breadcrumb docs (#474)
telerik/blazor-docs: components/breadcrumb/navigation.md

---
title: Navigation
page_title: Breadcrumb - Navigation
description: Using the Blazor Breadcrumb for navigating between pages.
slug: breadcrumb-navigation
tags: telerik,blazor,breadcrumb,navigation
published: True
position: 7
---

# Breadcrumb for Navigation

The Breadcrumb can be used to navigate between different pages in the application. It can generate the needed links for you through its `UrlField` when [data binding]({%slug breadcrumb-data-binding%}).

> In some cases you might prefer to dynamically generate the breadcrumbs based on the current application `Url`. For this scenario you can use the [`NavigationManager`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.navigationmanager?view=aspnetcore-5.0) that the component supports and subscribe to its [`LocationChanged`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.navigationmanager.locationchanged?view=aspnetcore-5.0) event. Thus, you can set new Breadcrumb items on every location change.

Reviewer: Do we need to hard-code aspnetcore-5.0 in the links? They work without this parameter.

Author: Actually we don't, I removed them.

>caption To use the Breadcrumb and dynamically generate its items:

* Add the Breadcrumb to your application.
    * You may want to add it in the `MainLayout.razor` outside of the `@Body`, for example, in the `main` container of your app.

> * Introduce a `NavigationManager` instance by injecting it in the file you would like to build the breadcrumbs.

Reviewer: There is no need to suggest and action as a concept and then explain how to do it. Directly tell what to do.

```suggestion
* Inject a `NavigationManager` instance in the file where you want to build the breadcrumbs.
```

> * Create an event handler and subscribe it to the `LocationChanged` event of the `NavigationManager` to handle every change in the location.

Reviewer: The same idea applies here, for example:

```suggestion
* Handle the `LocationChanged` event of the `NavigationManager`.
```

* Dynamically generate the Breadcrumb items based on the `Url` provided by the `Location` field of the `LocationChangedEventArgs`.

> Here is an example for such a configuration in our public repository - [SubScribeToLocationChanged]().

Author: #https://github.com/telerik/blazor-docs/pull/474#pullrequestreview-750325667
I believe you are referring to this missing link. If so, it is intentional - the link should point to a sample app in the Blazor UI repo which will be added shortly after the release and then I can plug the link. The app cannot be added prior to the release as the component is still not live. I want to add the link afterwards to make sure it will be the correct one.
Shall we proceed so or there could be a better solution for the case (e.g. include the expected link and after the release double check if correct)?

Author: Included the supposed link for the sample project.

> An example of such a configuration you may find in our public repository - [SubScribeToLocationChanged]().

Reviewer: Missing link?

```suggestion
Here is an example of such a configuration in our public repository - [SubScribeToLocationChanged]().
```
